cartoframes/dataset.py

```diff
@@ -26,25 +35,33 @@ class Dataset(object):
 
     DEFAULT_RETRY_TIMES = 3
 
-    def __init__(self, carto_context, table_name, schema='public', df=None):
-        self.cc = carto_context
-        self.table_name = normalize_name(table_name)
+    def __init__(self, query, data_type, context=None, bounds=None, schema='public', df=None):
+        self.cc = context or default_context
+        self.query = query
+        self.type = data_type
+        self.bounds = bounds
+        self.table_name = ''
+        # self.table_name = normalize_name(table_name)
         self.schema = schema
         self.df = df
         if self.df is not None:
             self.normalized_column_names = _normalize_column_names(self.df)
-        warn('Table will be named `{}`'.format(table_name))
+        # warn('Table will be named `{}`'.format(table_name))
 
     @staticmethod
-    def create_from_query(cart_context, query, table_name):
-        dataset = Dataset(cart_context, table_name)
-        dataset.cc.batch_sql_client \
-               .create_and_wait_for_completion(
-                   '''BEGIN; {drop}; {create}; {cartodbfy}; COMMIT;'''
-                   .format(drop=dataset._drop_table_query(),
-                           create=dataset._create_table_from_query(query),
-                           cartodbfy=dataset._cartodbfy_query()))
+    def create_from_table(table_name, context=None):
+        dataset = Dataset('SELECT * FROM {}'.format(table_name), 'Query', context)
+        return dataset
 
+    @staticmethod
+    def create_from_query(query, context=None):
+        dataset = Dataset(query, 'Query', context)
```

Comment:
If using the `@classmethod` decorater, this would be rewritten as:
```python
def create_from_query(cls, query, context=None):
    dataset = cls(query, 'Query', context)
```

Reply:
Yes, this is @oleurud terrain, but I can apply the changes